# Local patch (no upstream PR): ICONV_CONST is normally defined by
# autotools during configure, which we don't run. Hack in platform-specific
# logic instead: const on Windows and Solaris, empty elsewhere.
diff --git a/src/readstat_iconv.h b/src/readstat_iconv.h
index 38fb981..1327fc8 100644
--- a/src/readstat_iconv.h
+++ b/src/readstat_iconv.h
@@ -1,10 +1,10 @@
 #include <iconv.h>
 
-/* ICONV_CONST defined by autotools during configure according
- * to the current platform. Some people copy-paste the source code, so
- * provide some fallback logic */
-#ifndef ICONV_CONST
-#define ICONV_CONST
+/* ICONV_CONST defined by autotools; so we hack this in manually */
+#if defined(_WIN32) || defined(__sun)
+  #define ICONV_CONST const
+#else
+  #define ICONV_CONST
 #endif
 
 typedef ICONV_CONST char ** readstat_iconv_inbuf_t;
